heat/core/logical.py

```diff
@@ -519,6 +519,9 @@ def sanitize_input_type(
     # if one of the tensors is distributed, unsplit/gather it
     if x.split is not None and y.split is None:
         t1 = manipulations.resplit(x, axis=None)
```

Comment:
@samadpls `resplit(x, axis=None)` collects all distributed slices of `x` and creates a copy of the entire `x` on each process, this is what we want to avoid.